### PR TITLE
added post_date_gmt and post_modified_gmt manually as they were not set automatically

### PR DESCRIPTION
**Pull request by NAbdulla1**
The problem caused is because wp_insert_post was not setting post_date_gmt and post_modified_gmt automatically.

fixes this [Dokan Pro issue](https://github.com/weDevsOfficial/dokan-pro/issues/1711)

### CODE REVIEW

File: includes/Dashboard/Templates/Products.php [hunk 1]
```diff
@@ -268,13 +268,16 @@ public function handle_product_add() {
             self::$errors = apply_filters( 'dokan_can_add_product', $errors );
 
             if ( ! self::$errors ) {
+                $timestamp      = time();
```
Comment (nurul-umbhiya) [inline includes/Dashboard/Templates/Products.php:271]: change this line with:
```
$timenow = dokan_current_datetime()->setTimezone( new DateTimeZone( 'UTC' ) );
```

File: includes/Dashboard/Templates/Products.php [hunk 2]
```diff
@@ -268,13 +268,16 @@ public function handle_product_add() {
...
-                    'post_title'   => $post_title,
-                    'post_content' => $post_content,
-                    'post_excerpt' => $post_excerpt,
+                $post_data      = apply_filters( 'dokan_insert_product_post_data', array(
+                    'post_type'         => 'product',
+                    'post_status'       => $product_status,
+                    'post_title'        => $post_title,
+                    'post_content'      => $post_content,
+                    'post_excerpt'      => $post_excerpt,
+                    'post_date_gmt'     => date( 'Y-m-d H:i:s', $timestamp ),
```
Comment (nurul-umbhiya) [inline includes/Dashboard/Templates/Products.php:279]: change this line with: 
```
$timenow->format( 'Y-m-d H:i:s ),
```

File: includes/Dashboard/Templates/Products.php [hunk 3]
```diff
@@ -268,13 +268,16 @@ public function handle_product_add() {
...
+                    'post_modified_gmt' => date( 'Y-m-d H:i:s', $timestamp ),
```
Comment (nurul-umbhiya) [inline includes/Dashboard/Templates/Products.php:280]: change this line with:
```
$timenow->format( 'Y-m-d H:i:s ),
```

**Review by nurul-umbhiya — APPROVED** [review]

File: includes/Dashboard/Templates/Products.php (hunk 1 above)
Comment (nurul-umbhiya) [reply includes/Dashboard/Templates/Products.php:271]: Also, do we need to think about existing entries? If so, we are including a upgrader in this release.

**Review by nurul-umbhiya — CHANGES_REQUESTED** [review]

File: includes/Dashboard/Templates/Products.php (hunk 1 above)
Comment (NAbdulla1) [reply includes/Dashboard/Templates/Products.php:271]: yes vai, there can be a migrator to fix existing DB entries. possible approach: copy values from `post_date` and `post_modified` columns to `post_date_gmt` and `post_modified_gmt` with proper timezone conversion.

**Comment by NAbdulla1** [comment]
yes vai, there can be a migrator to fix existing DB entries. possible approach: copy values from `post_date` and `post_modified` columns to `post_date_gmt` and `post_modified_gmt` respectively with proper timezone conversion.

**Review by nurul-umbhiya — APPROVED** [review]

**Pull request merged**